### DISCUSSION TITLE
How could I set width in div with subfigures?

### DISCUSSION THREAD

**Discussion by aleixalcacer** [quarto-dev/quarto-cli#7475] [Q&A]
### Description

Hi,

I want to create a figure with two subfigures inside but I would need to set the width of the figure to 66%. I have tried many things but I have not been able to modify the width. I would like to get something like the following:

```qmd
::: {#fig-figure layout-ncol=2 width=66%}

![Subfigure 1](figure_1.png){#fig-sub1 width=90%}

![Subfigure 2](figure_1.png)(#fig-sub2 width=90%}

Figure caption.
:::
```

Is there any way to get it?

Thank you so much,
Aleix

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/authoring/figures.html#complex-layouts>.

**Reply by aleixalcacer** [reply]
Thank you so much :)

What do you recommmend me to do? Maybe, something like the following?

```
::: {#fig-figure layout="[[-16.66, 33, 33, -16.66]]"}

![Subfigure 1](figure_1.png){#fig-sub1 width=90%}

![Subfigure 2](figure_1.png)(#fig-sub2 width=90%}

Figure caption.
:::
```

However, do you know if there is any other solution (without having to do the trick of putting the -16.66)?

**Reply by mcanouil** [reply]
I don't know what layout you want, especially because you also change the width of the figures within the columns.
The layout is up to you, there are no recommendation.

`-X` is not a "trick", it's a feature to add spacing.

Not that the values in the layout are relatives, so you don't need to use decimal values.

````qmd
---
title: "Quarto Playground"
format: html
---

This is a playground for Quarto.

::: {#fig-figure layout="[[-1, 2, 4, -1]]"}

![Subfigure 1](https://placehold.co/200x200){#fig-sub1}

![Subfigure 2](https://placehold.co/400x200){#fig-sub2}

Figure caption.
:::
````

<img width="729" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/87654f98-2cb5-4992-b5c8-7522841e05fb">

**Reply by aleixalcacer** [reply]
Ok, thank you. Now I understand better the syntax :)

One last question, instead of using `layout="[[-1, 1, 1, -1]]"`, could I use `layout-ncol=2` and add space on the edges using some command? Or is that not possible? Because, for example, if I had more than 10 images, the first syntax could be very cumbersome.

Anyway, thank you very much for the clarification!

**Reply by mcanouil** [reply]
Ncol means equal widths columns.
It means you can't and have to specify the complete layout.
`layout="[[-1, 1, 1, -1], [-1, 1, 1, -1]]"` for 4 figures.
Notes that since you want space around, you could simply put le layout div inside a div with width. (Not sure it will properly work in LaTeX.)
